Makefile

```diff
@@ -13,4 +17,27 @@ latest:
 .PHONY: clean
 clean:
 	docker rmi -f $(IMAGE_NAME)
+	docker rmi `docker images --filter "label=intermediate=true" -q`
+
+.PHONY: test test-gen test-check test-clean
+test: test-gen test-check test-clean
+
+test-gen:
```

Comment:
does it make sense to consider adding some black box tests for gentool? we can think of adding logic that will compare generated files with some predefined result.

Reply:
Definitely something to consider, but requires some discussion on the best way to implement.

Reply:
Let's move this into an issue for future work -- I'd really love to get this smaller image out the door to see if I can use if for toolkit integration tests

Reply:
As a disclaimer (I mentioned this to Dan in person), the final image is small ~100MB, but the intermediate container is huge, about 2GB.